rskj-core/src/main/java/co/rsk/net/handler/txvalidator/TxValidatorStep.java

```diff
@@ -32,6 +33,7 @@
  */
 public interface TxValidatorStep {
 
-    TransactionValidationResult validate(Transaction tx, @Nullable AccountState state, BigInteger gasLimit, Coin minimumGasPrice, long bestBlockNumber, boolean isFreeTx);
+    TransactionValidationResult validate(Transaction tx, @Nullable AccountState state, BigInteger gasLimit, Coin minimumGasPrice, long bestBlockNumber, boolean isFreeTx, ClaimTransactionInfoHolder claimTransactionInfoHolder);
```

Comment:
I think add this method to be implemented to all other ones just because of a parameter it's too cumbersome. You had to change all validators just to change the logic of one one them.

Wouldn't be possible `TxValidatorAccountBalanceValidator` to receive the parameters necessary to create a copy of the ClaimTransactionInfoHolder? This way, you wouldn't need to pass the object every time you are calling `shouldAcceptTx`?

Reply:
At first I created a constructor to send the parameters like you mentioned but this would work in the end because some parameter like RepositorySnapshot might change after the `TxValidatorAccountBalanceValidator` instance is created. That is why I suggested overloading the validate method to add the ClaimTransactionInfoHolder wrapper, that way I can pass the most recent values of the arguments that might change.